Modules/SubModule_Import/xASL_imp_NII2BIDS.m

```diff
@@ -70,18 +74,43 @@
 	% Go through all subjects
 	listSubjectsSessions = xASL_adm_GetFileList(imPar.TempRoot,[],false,[],true);
     for iSubjectSession = 1:length(listSubjectsSessions)
-        x = xASL_imp_NII2BIDS_Subject(x,imPar,bidsPar,studyPar,listSubjectsSessions{iSubjectSession});
+        % Only run it for the current subject (maybe we can do this more elegantly in the future)
+        if ~isempty(regexpi(listSubjectsSessions{iSubjectSession},subjectName))
+            x = xASL_imp_NII2BIDS_Subject(x,imPar,bidsPar,studyPar,listSubjectsSessions{iSubjectSession});
+        end
+    end
+    
+    % Move log files of current subject from datasetroot & temp to derivatives/ExploreASL if they aren't there already
+    logFilesDatasetRoot = xASL_adm_GetFileList(x.dir.DatasetRoot,'^import.+$');
+    logFilesTempRoot = xASL_adm_GetFileList(imPar.TempRoot,'^import.+$');
+    allLogFiles = vertcat(logFilesDatasetRoot,logFilesTempRoot);
+    if ~isempty(allLogFiles)
+        for importFile=1:size(allLogFiles,1)
+            % Check for subject name
+            if ~isempty(regexpi(allLogFiles{importFile},subjectName))
+                [~,thisFileMeta,thisExtensionMeta] = xASL_fileparts(allLogFiles{importFile,1});
+                xASL_Move(allLogFiles{importFile,1},fullfile(imPar.DerivativesRoot,'ExploreASL',[thisFileMeta thisExtensionMeta]),1);
+            end
+        end
+    end
+    
+    % Delete temp folder of current subject
+    tempDirs = xASL_adm_GetFileList(imPar.TempRoot,[],[],[],true);
+    for iTempDir=1:numel(tempDirs)
+        if ~isempty(regexpi(tempDirs{iTempDir},subjectName))
+            xASL_delete(fullfile(tempDirs{iTempDir}), true);
+        end
     end
     
-    % Copy log files
-    importMetaFiles = xASL_adm_GetFileList(imPar.TempRoot,'^import.+$');
-    for importFile=1:size(importMetaFiles,1)
-        [~,thisFileMeta,thisExtensionMeta] = xASL_fileparts(importMetaFiles{importFile,1});
-        xASL_Copy(importMetaFiles{importFile,1},fullfile(studyPath,[thisFileMeta thisExtensionMeta]));
+    % Delete temp directory if it is empty
+    dirsInTemp = xASL_adm_GetFileList(imPar.TempRoot,[],[],[],true);
+    filesInTemp = xASL_adm_GetFileList(imPar.TempRoot,[],[],[],false);
+    if isempty(dirsInTemp) && isempty(filesInTemp)
+        xASL_delete(imPar.TempRoot, true);
     end
     
-    % Delete temp folder
-    xASL_delete(imPar.TempRoot, true);
+    % Update x.opts.DatasetRoot
```

Comment:
Can we use the content of `xASL_imp_UpdateDatasetRoot` also for the first time we determine the `DatasetRoot`? E.g., not only update but rather `xASL_imp_GetDatasetRoot`? To simplify the code.

Reply:
Would move this to further optimize in a new issue.

Reply:
Sure if it was there already, but I thought you just created this